internal/sdkprovider/service/kafkatopic/kafka_topic_exists.go

```diff
@@ -0,0 +1,40 @@
+package kafkatopic
+
+import (
+	"context"
+	"log"
+	"sync"
+
+	"github.com/aiven/aiven-go-client/v2"
+	"golang.org/x/exp/slices"
+)
+
+var onceForService sync.Map
+
+// isTopicExists checks if topic exists
+func isTopicExists(ctx context.Context, client *aiven.Client, project, serviceName, topic string) bool {
+	c := getTopicCache()
+
+	var err error
+	once, _ := onceForService.LoadOrStore(project+"/"+serviceName, new(sync.Once))
+	once.(*sync.Once).Do(func() {
+		var list []*aiven.KafkaListTopic
+		list, err = client.KafkaTopics.List(ctx, project, serviceName)
+		if err != nil {
+			return
+		}
+
+		c.SetV1List(project, serviceName, list)
+	})
+
+	if err != nil {
+		log.Printf("[ERROR] cannot check kafka topic existence: %s", err)
+		return false
+	}
+
+	if slices.Contains(c.GetV1List(project, serviceName), topic) || slices.Contains(c.GetQueue(project, serviceName), topic) {
```

Comment:
for better performance, if the topic is found in the first `c.GetV1List` check, there's no need to check in the second `c.GetQueue` call: you can return true immediately

Reply:
It does not make any sense to me, `slices.Contains` takes nano seconds and since we have `OR ||` operation, interpreter first check the `c.GetV1List` and if it is `true` second checks will be spiked by interpreter.

Reply:
please see this code example: https://github.com/aiven/terraform-provider-aiven/pull/1410#discussion_r1369798811

this approach allows to reduce to complexity by moving the second `slices.Contains` call to the `return` statement

Reply:
Ok, now I see your point will change that.